tests/std/tests/P0218R1_filesystem/test.cpp

```diff
@@ -994,12 +994,10 @@ void test_directory_entry() {
         EXPECT(!nonexistentEntry.exists());
 
         directory_entry nonexistentEntryEc(nonexistent, ec);
-        EXPECT(nonexistentEntryEc.path() == nonexistent);
+        EXPECT(nonexistentEntryEc.path() == path{});

```

Comment:
While the changes to the `error_code` overloads in `<filesystem>` appear to be correct, the behavior as depicted in this test is surprising. Specifically, it is strange that the non-`error_code` construction can succeed, while the `error_code` construction fails and produces different state. The general behavior of `<filesystem>` should be that the exception interface and `error_code` interface behave identically, except for how they report errors.

(And if the non-`error_code` overloads are changed to call `refresh()` as specified by the Standard, they would begin throwing - which would restore consistency, but would result in even greater changes in observable behavior.)

I think this reduces down to a question about how `refresh()`/`refresh(ec)` should behave for nonexistent paths. Why do they report an error, instead of successfully caching the fact that the path is non-existent? Is it because the Windows API makes it hard/impossible to distinguish "the path does not exist" versus "we couldn't determine what the path resolved to"? We have:

https://github.com/microsoft/STL/blob/68b344c9dcda6ddf1a8fca112cb9033f9e50e787/stl/inc/xfilesystem_abi.h#L51-L61

, is that relevant?